src/kpatch_coro.c

```diff
@@ -562,41 +574,34 @@ int kpatch_init_coroutine(struct kpatch_process *proc)
 	return 0;
 }
 
-int kpatch_find_coroutines(struct kpatch_process *proc)
+int kpatch_coroutines_find(struct kpatch_process *proc)
 {
-	if (!proc->coro.ops)
+	int i, rv;
+
+	/* Freshly started binary can't have coroutines */
+	if (proc->is_just_started)
 		return 0;
-	return proc->coro.ops->find_coroutines(proc);
+
+	for (i = 0; i < ARRAY_SIZE(kpatch_coro_flavours); i++) {
+		struct kpatch_coro_ops *ops = &kpatch_coro_flavours[i];
+
+		rv = ops->find_coroutines(proc);
+		if (rv == CORO_SEARCH_NEXT)
+			continue;
+
+		return rv;
+	}
```

Comment:
I'm thinking, may we should do here something like this:

if (rc == CORO_SEARCH_NEXT && process_is_known_to_use_coroutines()) {
  log("Warning: process contains coroutines but kcare doesn't know how to handle them")
  if (!force)
    exit()
}

And keep a list of processes we know for sure use coroutines, like qemu.

Reply:
Updated, please review.

Reply:
I will switch to checking whether `makecontext`, `setjmp` and so on are used by the process.